### PR TITLE
PXC-3292: Enable the whole MTR test suite for PXC 5.7

### PR DESCRIPTION
**Pull request by kamil-holubicki**
https://jira.percona.com/browse/PXC-3292

Enabled all MTR tests. Fixes done:
1. MTR framework Perl scripts fixed
2. Tests results rerecorded because of output change caused by PXC additions
3. Tests results rerecorded because of binlog_format default changed to ROW in PXC
4. Disabled tests that use variables that are read only in PXC
5. PFS_MAX_STAGE_CLASS changed 150 -> 180 to allow PXC related instrumentation
6. Group replication tests fixed. This involved adjusting code logic in lock0lock.cc and rpl_gtid_state.cc
7. Fixed test percona_rpl_mts_backup_lock_ps3345 (change in binlog.cc)
8. -D_GLIBCXX_ASSERTIONS disabled for CentOS 6/7 debug build (change in Galera repo)

Note that several (groups of) tests still fail:
PXC-3295 - HP transactions related tests
PXC-3323 - percona_extended_innodb_status.test
PXC-3297 - keyring_vault related tests
PXC-3296 - merge.test and merge_mmap.test
PXC-3283 - innodb-autoinc.test

### CODE REVIEW

**Comment by kamil-holubicki** [comment]
I will squash before merge. Left as separate commits to make it easier to review.

File: mysql-test/r/mysqldump.result
Comment (dutow) [inline mysql-test/r/mysqldump.result:682]: This is some accidental change, maybe because of the editor encoding?

File: sql/rpl_gtid_state.cc
Comment (dutow) [inline sql/rpl_gtid_state.cc]: typo: server_sidno

File: storage/innobase/lock/lock0lock.cc
Comment (dutow) [inline storage/innobase/lock/lock0lock.cc]: High priority transactions also work with codership mysql, I based the 8.0 fix on that. We could use the same patch here.

File: mysql-test/r/mysqldump.result
Comment (kamil-holubicki) [reply mysql-test/r/mysqldump.result:682]: This is OK. '??' is wrong caused by editor. 
This file looks like:
image
and in fact is the same as upstream (PS) version of it.

File: storage/innobase/lock/lock0lock.cc
Comment (kamil-holubicki) [reply storage/innobase/lock/lock0lock.cc]: Hi @venkatesh-prasad-v , can you please check it out?

Comment (venkatesh-prasad-v) [reply storage/innobase/lock/lock0lock.cc]: Hi @dutow!! I and Kamil decided to handle HPTs as part of PXC-3295. So, we have reverted the changes done in lock0lock.cc and will work towards evaluating the backport of the 8.0 fix to 5.7.

**Review by dutow — APPROVED** [review]
Approved after squashing.

**Pull request merged**
